modules/apps/portal-search/portal-search-web/src/main/resources/com/liferay/portal/search/web/internal/category/facet/portlet/display/template/dependencies/portlet_display_template_vocabulary.ftl

```diff
@@ -10,13 +10,15 @@
 >
 	<li class="treeview-item" role="none">
 		<#if name?has_content>
+			<#assign divRandomId = stringUtil.randomId() />
+
 			<div
 				aria-controls="${(termDisplayContexts?has_content)?then(namespace + 'treeItem' + id, '')}"
 				aria-expanded="true"
 				class="treeview-link ${cssClassTreeItem}"
 				data-target="#${namespace}treeItem${id}"
 				data-toggle="collapse"
-				onClick="${namespace}toggleTreeItem('${namespace}treeItem${id}');"
+				id="${divRandomId}"
 				onKeyPress="${namespace}toggleTreeItemKeypress(event);"
 				role="treeitem"
```

Comment:
Wondering if this onKeyPress has to be done as well?

Reply:
Oops, good catch :raised_hands:  ...

It looks like that event is deprecated -> https://www.w3schools.com/jsref/event_onkeypress.asp that's the reason why it didn't show in the spec, thus I didn't include `onkeypress` in my searches when adding the changes to be CSP compliant.

But it is true that even being deprecated it can probably break CSP so I leave up to you whether moving that handler to its own script node or use `onkeydown` + its own script node instead, as the documentation suggests.

Thx :-)

Reply:
I'm assigning the LPD to you again. Once that's done, you can forward this to Brian, as there's nothing else to be done on our side. Thx.

Reply:
Sounds good! Since it is deprecated, I'll leave it as is for now and we can send changes later if necessary.

Looks like the ci commands are working now so I'll try forwarding :) Thanks!